File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline
n = int(input())

arr = []

for _ in range(n):
    cmd = input().strip()
    if 'push' in cmd:
        arr.append(int(cmd[5:]))

    elif cmd == 'pop':
        if len(arr) == 0:
            print(-1)
        else:
            print(arr[-1])
            arr.pop(-1)

    elif cmd == 'size':
        print(len(arr))

    elif cmd == 'empty':
        if len(arr) == 0:
            print(1)
        else:
            print(0)

    elif cmd == 'top':
        if len(arr) == 0:
            print(-1)
        else:
            print(arr[-1])

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

    static Stack<Integer> stack;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());
        stack = new Stack<>();

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();

            if (cmd.equals("push")) {
                int num = Integer.parseInt(st.nextToken());
                stack.push(num);
            } else if (cmd.equals("pop")) {
                if (!stack.isEmpty()) {
                    System.out.println(stack.pop());
                } else {
                    System.out.println(-1);
                }
            } else if (cmd.equals("size")) {
                System.out.println(stack.size());
            } else if (cmd.equals("empty")) {
                System.out.println((stack.isEmpty()) ? 1 : 0);
            } else {
                if (!stack.isEmpty()) {
                    System.out.println(stack.peek());
                } else {
                    System.out.println(-1);
                }
            }
        }

        br.close();
    }


}

